src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/RazorGeneratedDocumentIdentityExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.NET.Sdk.Razor.SourceGenerators;
+
+namespace Microsoft.CodeAnalysis;
+
+internal static class RazorGeneratedDocumentIdentityExtensions
+{
+    public static bool IsRazorSourceGeneratedDocument(this RazorGeneratedDocumentIdentity identity)
+    {
+        return identity.GeneratorTypeName == typeof(RazorSourceGenerator).FullName;
```

Comment:
When would we ever have a `RazorGeneratedDocumentIdentity` that wasn't from our source generator?

Reply:
When I wrote this, yeah, we could get them when, for example, you renamed a method from a Razor file, where the method was defined in C# and generated by the System.Text.Json or Regex generators. I've made recent changes in Roslyn to rename that prevents that now, but it wouldn't take much change on the Roslyn side for that to be undone.